vars/rpmDashboardsDistValidation.groovy

```diff
@@ -0,0 +1,127 @@
+    def version = bundleManifestObj.build.version        //1.3.0
+    def architecture = bundleManifestObj.build.architecture
+    def plugin_names = bundleManifestObj.getNames();
+
+    def latestOpenSearchURL = "https://ci.opensearch.org/ci/dbc/Playground/tianleh-test/tianle-opensearch-build-3-22/$version/latest/linux/$architecture/rpm/dist/opensearch/opensearch-$version-linux-${architecture}.rpm"
+    def latestOpensearchDist = "$WORKSPACE/opensearch-$version-linux-${architecture}.rpm"
+    // Download the latest OpenSearch distribution with same version to be compatible with Dashboards.
+    sh("curl -SLO $latestOpenSearchURL")
+
+    // This is a reference meta data which the distribution should be consistent with.
+    def refMap = [:]
+    refMap['Name'] = name
+    refMap['Version'] = version
+    refMap['Architecture'] = architecture
+    refMap['Group'] = "Application/Internet"
+    refMap['License'] = "Apache-2.0"
+    refMap['Relocations'] = "(not relocatable)"
+    refMap['URL'] = "https://opensearch.org/"
+    // The context of meta data should be for OSD
+    refMap['Summary'] = "Open source visualization dashboards for OpenSearch"
+    refMap['Description'] = "OpenSearch Dashboards is the visualization tool for data in OpenSearch\n" +
+            "For more information, see: https://opensearch.org/"
+
+    //Validation for the Meta Data of distribution
+    println("Meta data validations start:")
+    def metadata = sh (
+            script: "rpm -qip $distFile",
+            returnStdout: true
+    ).trim()
+    println("Meta data for the RPM distribution is: \n" + metadata)
+    // Extract the meta data from the distribution to Map
+    def metaMap = [:]
+    for (line in metadata.split('\n')) {
+        def key = line.split(':')[0].trim()
+        if (key != 'Description') {
+            metaMap[key] = line.split(':', 2)[1].trim()
+        } else {
+            metaMap[key] = metadata.split(line)[1].trim()
+            break
+        }
+    }
+    // Start validating
+    refMap.each{ key, value ->
+        if (key == "Architecture") {
+            if (value == 'x64') {
+                assert metaMap[key] == 'x86_64'
+            } else if (value == 'arm64') {
+                assert metaMap[key] == 'aarch64'
+            }
+        } else {
+            assert metaMap[key] == value
+        }
+        println("Meta data for $key is validated")
+    }
+    println("Validation for meta data of RPM distribution completed.")
+
+    //Validation for the installation
+    //Install the rpm distribution via yum
+    println("Start installation with yum.")
+    sh "sudo yum install -y $distFile"
+    println("RPM distribution for $name is installed with yum.")
+    sh "sudo yum install -y $latestOpensearchDist"
+    println("Latest RPM distribution for OpenSearch is also installed with yum.")
+
+    //Start the installed OpenSearch-Dashboards distribution
+    sh ("sudo systemctl restart $name")
+    sleep 30
```

Comment:
Is there a better way to do this than just wait long enough? try to hit the endpoint in a loop in a `waitForService [url]` function? You could simplify both waiting for OS and OSD by just giving a URL and looking for certain text with a timeout.

Reply:
I will investigate on this suggestion

Reply:
You can refer to this example https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/service.py#L83